python/lsst/meas/algorithms/sourceSelector.py

```diff
@@ -97,3 +103,301 @@ def _isBad(self, source):
 sourceSelectorRegistry = pexConfig.makeRegistry(
     doc="A registry of source selectors (subclasses of BaseSourceSelectorTask)",
 )
+
+
+class ColorLimit(pexConfig.Config):
```

Comment:
I think these configs and the new SourceSelector below probably belong in their own `.py` file.

Also, I feel like the names of the new configs should be e.g. `ColorLimitConfig`?

Reply:
Maybe they belong in their own file, but I don't know what to call that file except `sourceSelector.py`!

I've a mild preference for not automatically adding `Config` to the end of the names: it's implemented as a `Config` (which is useful for configuring them) but their principal purpose is as a limit. If you have strong feelings, let me know and we'll hash it out.

Reply:
In that case, I wonder if the we should instead move the existing `BaseSourceSelector` into its own eponymous file then?

I don't have a strong feeling about appending `Config`, but the names as they are just feel short somehow.